src/control/server/storage/scm/ipmctl_dimm_test.go

```diff
@@ -0,0 +1,234 @@
+  </Dimm>`
+	sockOneOutList := `<DimmList>` + sockOneOut + `</DimmList>`
+	genCmdOutputMap := func() OutputMap {
+		return OutputMap{
+			cmdShowIpmctlVersion.String(): `
+Intel(R) Optane(TM) Persistent Memory Command Line Interface Version 03.00.00.0468`,
+			cmdShowDIMMs.String(): `
+<?xml version="1.0"?>
+ <DimmList>
+  <Dimm>
+   <DimmID>0x0001</DimmID>
+   <Capacity>502.599 GiB</Capacity>
+   <HealthState>Healthy</HealthState>
+   <FWVersion>01.00.00.5127</FWVersion>
+   <PhysicalID>0x001e</PhysicalID>
+   <DimmUID>8089-a2-1839-000010ce</DimmUID>
+   <SocketID>0x0000</SocketID>
+   <MemControllerID>0x0000</MemControllerID>
+   <ChannelID>0x0000</ChannelID>
+   <ChannelPos>1</ChannelPos>
+   <PartNumber>NMA1XXD512GQS</PartNumber>
+  </Dimm>
+  <Dimm>
+   <DimmID>0x0101</DimmID>
+   <Capacity>502.599 GiB</Capacity>
+   <HealthState>Healthy</HealthState>
+   <FWVersion>01.00.00.5127</FWVersion>
+   <PhysicalID>0x0024</PhysicalID>
+   <DimmUID>8089-a2-1839-000010e7</DimmUID>
+   <SocketID>0x0000</SocketID>
+   <MemControllerID>0x0001</MemControllerID>
+   <ChannelID>0x0000</ChannelID>
+   <ChannelPos>1</ChannelPos>
+   <PartNumber>NMA1XXD512GQS</PartNumber>
+  </Dimm>
+` + sockOneOut + `</DimmList>`,
+		}
+	}
+	one := 1
+
+	for name, tc := range map[string]struct {
+		sockSelector *int
+		cmdOutputMap OutputMap
+		cmdErrorMap  ErrorMap
+		expModules   storage.ScmModules
+		expErr       error
+		expCalls     CallMap
+	}{
+		"invalid xml": {
+			cmdOutputMap: func() OutputMap {
+				om := genCmdOutputMap()
+				om[cmdShowDIMMs.String()] = `text that is invalid xml`
+				return om
+			}(),
+			expErr: errors.New("parse show dimm cmd"),
+		},
+		"no permissions": {
+			cmdErrorMap: func() ErrorMap {
+				em := genCmdErrorMap()
+				em[cmdShowDIMMs.String()] = errors.Wrap(&system.RunCmdError{
+					Wrapped: &exec.ExitError{
+						ProcessState: &os.ProcessState{},
+					},
+					Stdout: "Sorry, the " + outNoCLIPerms,
+				}, cmdShowDIMMs.String())
+				return em
+			}(),
+			expErr: errors.Errorf("ipmctl show -o nvmxml -d DimmID,ChannelID,ChannelPos,MemControllerID,SocketID,PhysicalID,Capacity,DimmUID,PartNumber,FWVersion,HealthState -dimm: exit status 0: stdout: Sorry, the %s; stderr: ", outNoCLIPerms),
+		},
+		"ipmctl version command fails": {
+			cmdErrorMap: func() ErrorMap {
+				em := genCmdErrorMap()
+				em[cmdShowIpmctlVersion.String()] = errors.New("fail version")
+				return em
+			}(),
+			expErr: errors.New("fail version"),
+		},
+		"show dimms command fails": {
+			cmdErrorMap: func() ErrorMap {
+				em := genCmdErrorMap()
+				em[cmdShowDIMMs.String()] = errors.Wrap(&system.RunCmdError{
+					Wrapped: &exec.ExitError{
+						ProcessState: &os.ProcessState{},
+					},
+				}, cmdShowDIMMs.String())
+				return em
+			}(),
+			expErr: errors.New("ipmctl show -o nvmxml -d DimmID,ChannelID,ChannelPos,MemControllerID,SocketID,PhysicalID,Capacity,DimmUID,PartNumber,FWVersion,HealthState -dimm: exit status 0: stdout: ; stderr: "),
```

Comment:
This is an unhelpful error if the command failed.

Reply:
This case was to verify that the RunCmdError is formatted as expected in the output error string. Specifically that the exit code is printed from the ExitError.ProcessedState. I am unable to supply a nonzero status as the Field is not exported in ProcessState but we can verify that it is printed in the output error.